lms/djangoapps/courseware/masquerade.py

```diff
@@ -326,6 +326,20 @@ def is_masquerading_as_specific_student(user, course_key):
     return bool(course_masquerade and course_masquerade.user_name)
 
 
+def get_specific_masquerading_user(user, course_key):
+    """
+    Return the specific user that a staff member is masquerading as, or None if they aren't.
+    """
+    course_masquerade = get_course_masquerade(user, course_key)
+    is_specific_user = bool(course_masquerade and course_masquerade.user_name)
+    if is_specific_user:
+        return CourseEnrollment.objects.users_enrolled_in(course_key).get(
+            username=course_masquerade.user_name
+        )
```

Comment:
Why raise an exception if the user isn't enrolled? I would have expected this to just be a lookup in the user table.

Reply:
This is the same logic as https://github.com/edx/edx-platform/pull/24588/files/046ba8a9f805d07322d378be58daf28a8fb7ee9d#diff-2ad237e74ccb2af44c5817cc78d4a732L203-L211.

That said, I think you're right, it's easier to just get the user directly, at this point in the code.

Reply:
I'm not sure that link is right?